fastcov stuck after json.decoder.JSONDecodeError exception

Hi,

Great program. Keep up the good work. We've added it to our nightly and last night it got stuck with this call stack:

11:52:27 [13.835s] [info]: Found 3034 coverage files (gcda)
11:52:27 [0.261s] [info]: Spawned 6 gcov processes, each processing at most 506 coverage files
12:05:55 Process Process-2:
12:05:55 Traceback (most recent call last):
12:05:55   File "/usr/lib64/python3.6/multiprocessing/process.py", line 258, in _bootstrap
12:05:55     self.run()
12:05:55   File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run
12:05:55     self._target(*self._args, **self._kwargs)
12:05:55   File "/usr/local/lib/python3.6/site-packages/fastcov.py", line 135, in gcovWorker
12:05:55     intermediate_json = json.loads(line.decode(sys.stdout.encoding))
12:05:55   File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
12:05:55     return _default_decoder.decode(s)
12:05:55   File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode
12:05:55     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
12:05:55   File "/usr/lib64/python3.6/json/decoder.py", line 355, in raw_decode
12:05:55     obj, end = self.scan_once(s, idx)
12:05:55 json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 69632 (char 69631)

I had to kill it manually to unstuck the nightly build. We are using gcc 9.2. Thanks.

Hello!

We have similar issue (occurs randomly):

```
Resetting code coverage counters to zero.
Processing code coverage counters and generating report.
/usr/local/bin/fastcov --zerocounters
[0.158s] [info]: Found 344 coverage files (gcda)
[0.055s] [info]: Removed 344 .gcda files
/usr/local/bin/fastcov --jobs 8 --exclude /usr/include /root/.conan test/ --process-gcno --lcov --output build/coverage.info.base
[0.082s] [info]: Found 502 coverage files (gcno)
[0.026s] [info]: Spawned 8 gcov processes, each processing at most 63 coverage files
Process Process-5:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.6/site-packages/fastcov.py", line 262, in gcovWorker
    intermediate_json = json.loads(line.decode(sys.stdout.encoding if sys.stdout.encoding else 'UTF-8'))
  File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 138846189 (char 138846188)
make[3]: *** [test/CMakeFiles/test_coverage.dir/build.make:83: test/CMakeFiles/test_coverage] Hangup
make[2]: *** [CMakeFiles/Makefile2:5510: test/CMakeFiles/test_coverage.dir/all] Hangup
make[1]: *** [CMakeFiles/Makefile2:5517: test/CMakeFiles/test_coverage.dir/rule] Hangup
make: *** [Makefile:2505: test_coverage] Hangup
```

Centos8
gcc version 9.2.1 20191120 (Red Hat 9.2.1-2)
gcov (GCC) 9.2.1 20191120 (Red Hat 9.2.1-2)
fastcov 1.8

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,7 +20,7 @@ script:
   - docker run -v "$TRAVIS_BUILD_DIR":/mnt/workspace -w /mnt/workspace rpgillespie6/fastcov bash -c "cd ./test && ./run_tests.sh"
 
 after_success:
-  - bash <(curl -s https://codecov.io/bash) -X gcov -X coveragepy -s $TRAVIS_BUILD_DIR # Upload coverage to codecov
+  - bash <(curl -s https://codecov.io/bash) -X coveragepy -s $TRAVIS_BUILD_DIR # Upload coverage to codecov
 
 deploy:
   provider: pypi

--- a/test/functional/bad-gcov.sh
+++ b/test/functional/bad-gcov.sh
@@ -1,0 +1,7 @@
+#!/bin/bash
+
+if [ "$1" == "-v" ]; then
+  echo "gcov (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0"
+fi
+
+echo "{bad json" # Pretend gcov crashed or was killed

--- a/test/functional/run_all.sh
+++ b/test/functional/run_all.sh
@@ -165,6 +165,11 @@ rc=0
 coverage run -a ${TEST_DIR}/fastcov.py -C ${TEST_DIR}/expected_results/missing_files.json --scan-exclusion-markers -o missing.json || rc=$?
 test $rc -eq 6
 
+# Test gcov getting killed or crashing midway through JSON output
+rc=0
+coverage run -a ${TEST_DIR}/fastcov.py --exceptional-branch-coverage --gcov ${TEST_DIR}/bad-gcov.sh --source-files ../src/source1.cpp --lcov -o multitest.actual.info || rc=$?
+test $rc -eq 7
+
 # Test (running data from other than build directory)
 RUN_DIR=${PWD}
 pushd ${TEST_DIR}

